src/BloomExe/Collection/CollectionSettings.cs

```diff
@@ -205,10 +206,10 @@ public string GetLanguage1Name(string inLanguage)
 
 		private string GetLanguage1Name_NoCache(string inLanguage)
 		{
-			Iso639LanguageCode exactLanguageMatch = _lookupIsoCode.GetExactLanguageMatch(Language1Iso639Code);
-			if (exactLanguageMatch == null)
+			var exactLanguageMatch = _lookupIsoCode.GetBestLanguageName(Language1Iso639Code);
+			if (exactLanguageMatch == Language1Iso639Code) // no useful name found
```

Comment:
So if a lang code = lang name, we're toast.  You mentioned this risk to the testers, but it really does seem... risky.